### PR TITLE
Add documentation about build secrets

### PR DESCRIPTION
**Pull request by petey**

### CODE REVIEW

File: docs/getting-started/secrets.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Build Secrets
+You've got secrets to share with your jobs, but these shouldn't be shared with everyone. Screwdriver provides a mechanism to insert secrets as environment variables. Since secrets are exposed as environment variables, they are easy to use inside builds.
+
+## Security
+The Screwdriver team takes security very seriously and encrypts all traffic between its various services. User secrets are stored encrypted in our datastore, and their values are only released to those builds that are authorized by the user.
+
+We understand that you, the security-conscious pipeline admin, may not wish to put secrets into pull-request builds, as a malicious pull-requester could expose those secrets without your consent, but still need those secrets as part of the main build. Screwdriver provides an additional flag on secrets, `allowInPR` (default: false), that is required to be enabled for a secret to be exposed.
+
+Secrets may only be added, modified, or removed by people that are listed as admins of the git repository associated with a given pipeline.
```
Comment (d2lam) [inline docs/getting-started/secrets.md]: Should we also mention that people with write can view secrets (just name, not value) that belong to a pipeline, or a build

File: docs/getting-started/secrets.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    "value": "MY_SECRET_VALUE!",
+    "allowInPR": false
+}
+```
+
+### Updating Secrets
+Send a PUT request to the screwdriver api server with appropriate security token headers:
+
+```
+POST /secrets/:id
```
Comment (minzcmu) [inline docs/getting-started/secrets.md]: Here should be `PUT`?

File: docs/getting-started/secrets.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+### Creating Secrets
+Simply enter the key and value in the inputs in the grey box, and click the add button. A checkbox is provided to allow you to enable `allowInPR`.
+
+### Updating Secrets
+A secret's original value is never delivered to the UI, but values of secrets may be updated in the UI by adding a new value in the text field next to the appropriate key name and clicking the update button.
+
+### Deleting secrets
+Individual secrets may be removed by clicking the Delete button.
+
+## Screwdriver API
```
Comment (minzcmu) [inline docs/getting-started/secrets.md:55]: And we also have `GET /secrets/:id`

**Review by minzcmu — CHANGES_REQUESTED** [review]

File: docs/getting-started/secrets.md (hunk 1 above)
Comment (petey) [reply docs/getting-started/secrets.md]: ok

File: docs/getting-started/secrets.md (hunk 3 above)
Comment (petey) [reply docs/getting-started/secrets.md:55]: I'll expose the fetch-all from pipeline. The fetch-single endpoint is less useful to end users.

File: docs/getting-started/secrets.md (hunk 2 above)
Comment (petey) [reply docs/getting-started/secrets.md]: yes

**Review by minzcmu — APPROVED** [review]

File: docs/getting-started/secrets.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        - publish-npm: npm publish
+    secrets:
+        # Publishing to NPM
+        - NPM_TOKEN
+```
+
+You may add secrets to any jobs you wish.
+
+### Secrets in pull-requests
+For your own security, Screwdriver doesn't recommend exposing secrets to pull-request builds. Pull-request jobs can be created with modified `screwdriver.yaml` files including changes to the secrets configuration.
```
Comment (tkyi) [inline docs/getting-started/secrets.md]: `For your own security, we don't`

File: docs/getting-started/secrets.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# Build Secrets
+You've got secrets to share with your jobs, but these shouldn't be shared with everyone. Screwdriver provides a mechanism to insert secrets as environment variables. Since secrets are exposed as environment variables, they are easy to use inside builds.
+
+## Security
+The Screwdriver team takes security very seriously and encrypts all traffic between its various services. User secrets are stored encrypted in our datastore, and their values are only released to those builds that are authorized by the user.
+
+We understand that you, the security-conscious pipeline admin, may not wish to put secrets into pull-request builds, as a malicious pull-requester could expose those secrets without your consent, but still need those secrets as part of the main build. Screwdriver provides an additional flag on secrets, `allowInPR` (default: false), that is required to be enabled for a secret to be exposed.
+
+Secrets may only be added, modified, or removed by people that are listed as admins of the git repository associated with a given pipeline. Admins may also fetch a list of secrets, but not the secret values.
```
Comment (tkyi) [inline docs/getting-started/secrets.md]: capitalize `Git`

**Review by stjohnjohnson — APPROVED** [review]
Love it! Merge after the tweaks from @tkyi

File: docs/getting-started/secrets.md (hunk 4 above)
Comment (petey) [reply docs/getting-started/secrets.md]: fixed.

File: docs/getting-started/secrets.md (hunk 5 above)
Comment (petey) [reply docs/getting-started/secrets.md]: fixed.

**Pull request merged**